reflex/utils/serializers.py

```diff
@@ -0,0 +1,180 @@
+"""Serializers used to convert Var types to JSON strings."""
+
+from __future__ import annotations
+
+import re
+from datetime import date, datetime, time, timedelta
+from typing import Any, Callable, Dict, Type, Union, get_type_hints
+
+from reflex.utils import exceptions, types
+
+# Mapping from type to a serializer.
+# The serializer should convert the type to a JSON object.
+Serializer = Callable[[Type], Union[str, dict]]
+SERIALIZERS: dict[Type, Serializer] = {}
+
+
+def serializer(fn: Serializer) -> Serializer:
+    """Decorator to add a serializer for a given type.
+
+    Args:
+        fn: The function to decorate.
+
+    Returns:
+        The decorated function.
+
+    Raises:
+        ValueError: If the function does not take a single argument.
+    """
+    # Get the global serializers.
+    global SERIALIZERS
+
+    # Check the type hints to get the type of the argument.
+    type_hints = get_type_hints(fn)
+    args = [arg for arg in type_hints if arg != "return"]
+
+    # Make sure the function takes a single argument.
+    if len(args) != 1:
+        raise ValueError("Serializer must take a single argument.")
+
+    # Get the type of the argument.
+    type_ = type_hints[args[0]]
+
+    # Make sure the type is not already registered.
+    registered_fn = SERIALIZERS.get(type_)
+    if registered_fn is not None and registered_fn != fn:
+        raise ValueError(
+            f"Serializer for type {type_} is already registered as {registered_fn}."
```

Comment:
`registered_fn.__qualname__` might be useful here, in case the name is ambiguous